integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java

```diff
@@ -550,4 +552,32 @@ public void seekAroundLocalBlock() throws IOException, TException {
     Assert.assertEquals(99, is.read());
     is.close();
   }
+
+  /**
+   * Test remote read stream lock in <code>RemoteBlockInStream</code>.
+   */
+  // TODO: figure out why this test fails
+  @Test
+  public void remoteReadLockTest()
+      throws IOException, InterruptedException, FileDoesNotExistException {
+    String uniqPath = PathUtils.uniqPath();
+    for (int k = MIN_LEN + DELTA; k <= MAX_LEN; k += DELTA) {
+      TachyonFile f =
+          TachyonFSTestUtils.createByteFile(mTfs, uniqPath + "/file_" + k, mWriteTachyon, k);
+
+      long blockId = mTfs.getInfo(f).getBlockIds().get(0);
+      BlockInfo info = TachyonBlockStore.get().getInfo(blockId);
+      RemoteBlockInStream is =
+          new RemoteBlockInStream(info.getBlockId(), info.getLength(), info.getLocations().get(0)
+              .getWorkerAddress());
+      Assert.assertEquals(0, is.read());
+      mTfs.delete(f);
+      Thread.sleep(1000);
```

Comment:
You can use `CommonUtils.sleepMs` and just sleep for 3 times the heartbeat interval.